neighsyncd/neighsyncd.cpp

```diff
@@ -27,16 +31,36 @@ int main(int argc, char **argv)
             NetLink netlink;
             Select s;
 
-            netlink.registerGroup(RTNLGRP_NEIGH);
-            cout << "Listens to neigh messages..." << endl;
-            netlink.dumpRequest(RTM_GETNEIGH);
+            using namespace std::chrono;
 
-            s.addSelectable(&netlink);
             if (sync.getRestartAssist()->isWarmStartInProgress())
             {
                 sync.getRestartAssist()->readTableToMap();
+
+                steady_clock::time_point starttime = steady_clock::now();
+                while (!sync.isNeighRestoreDone())
+                {
```

Comment:
Just some thought on current  restore_neighbors.py implementation.  For system warm reboot, the stale entries won't be filtered out and has to wait for a new cycle of arp age out time interval,  then the whole reconciliation logic in neighsyncd becomes redundant (skippable).  neighorch simply drops the duplicate entries and accept new entries.

Reply:
It is not exactly true, in case the neighbor ports were done during the warm reboot, the neighbor entries won't be inserted to kernel so we need reconciliation logic to remove them. 

Also it is good to use the same logic/code path to handle both scenarios. 

On top of that, we are still exploring the solutions to only get alive entries into kernel during the restoring phase if possible in later PRs.

So overall, we should use the same reconciliation logic.